### DISCUSSION TITLE
How to reduce the latency of a stream from the usb camera

### DISCUSSION THREAD

**Discussion by stefantripac** [SuRGeoNix/Flyleaf#487] [Q&A]
It is possible tor reduce the latency of the Flayleaf when Open stream camera with input string : "fmt://dshow?video=PC-LM1E&video_size=640x480&framerate=30" ?

After 10 minutes of working the lag varies from 1 to 18 secs. the bitrate  is 13,590 Kbps and codec : mjpeg, read from ShowDebug window.

Using VLC wit the same camera the result is almost the same,

The camera with other windows applications like Microsoft Teams work without any visible lag.

The camera options read with ffmpeg are:

[dshow @ 000001b59e6c9e40] DirectShow video device options (from video devices)
[dshow @ 000001b59e6c9e40]  Pin "Capture" (alternative pin name "Capture")
[dshow @ 000001b59e6c9e40]   vcodec=mjpeg  min s=1280x720 fps=30 max s=1280x720 fps=30
[dshow @ 000001b59e6c9e40]   vcodec=mjpeg  min s=1920x1080 fps=30 max s=1920x1080 fps=30
[dshow @ 000001b59e6c9e40]   vcodec=mjpeg  min s=800x600 fps=30 max s=800x600 fps=30
[dshow @ 000001b59e6c9e40]   vcodec=mjpeg  min s=640x480 fps=30 max s=640x480 fps=30
[dshow @ 000001b59e6c9e40]   vcodec=mjpeg  min s=352x288 fps=30 max s=352x288 fps=30
[dshow @ 000001b59e6c9e40]   vcodec=mjpeg  min s=320x240 fps=30 max s=320x240 fps=30
[dshow @ 000001b59e6c9e40]   pixel_format=yuyv422  min s=1920x1080 fps=5 max s=1920x1080 fps=5
[dshow @ 000001b59e6c9e40]   pixel_format=yuyv422  min s=1280x720 fps=10 max s=1280x720 fps=10
[dshow @ 000001b59e6c9e40]   pixel_format=yuyv422  min s=800x600 fps=20 max s=800x600 fps=20
[dshow @ 000001b59e6c9e40]   pixel_format=yuyv422  min s=640x480 fps=30 max s=640x480 fps=30
[dshow @ 000001b59e6c9e40]   pixel_format=yuyv422  min s=352x288 fps=30 max s=352x288 fps=30
[dshow @ 000001b59e6c9e40]   pixel_format=yuyv422  min s=320x240 fps=30 max s=320x240 fps=30

Can I do something in flyleaf configuration  to reduce this live stream lag ?

Thank you.

**Comment by SuRGeoNix** [comment]
Hi @stefantripac, yes there are a number of options for latency. Mainly set this:

```csharp
Config.Player.MaxLatency = TimeSpan.FromMilliseconds(500).Ticks;
```

You can search the issues / discussions for more info, here one related https://github.com/SuRGeoNix/Flyleaf/issues/388

**Reply by stefantripac** [reply]
It work , Thank you.
